src/Lecture2.hs

```diff
@@ -162,9 +181,53 @@ data Knight = Knight
     { knightHealth    :: Int
     , knightAttack    :: Int
     , knightEndurance :: Int
-    }
-
-dragonFight = error "TODO"
+    } deriving (Show)
+
+data Treasure = Treasure
+    { content :: String
+    } deriving (Show)
+
+data Chest = Chest
+    { treasure :: Maybe Treasure
+    , gold     :: Int
+    } deriving (Show)
+
+data Dragon = Dragon
+    { dragonFirePower :: Int
+    , dragonHealth    :: Int
+    , dragonType      :: DragonType
+    } deriving (Show)
+
+data DragonType = Red | Black | Green deriving (Enum, Eq, Show)
+
+getExperienceForDragon :: DragonType -> Int
+getExperienceForDragon dragonType
+                      | dragonType == Red = 100
+                      | dragonType == Black = 150
+                      | dragonType == Green = 250
```

Comment:
This function uses guards and that's why GHC produces a warning about non-exhaustive patterns ⚠️ 

It's better to use `case-of` here